include/ofi_mr.h

```diff
@@ -271,6 +271,7 @@ struct ofi_mr {
 	uint64_t flags;
 	enum fi_hmem_iface iface;
 	uint64_t device;
+	void *hmem_data;
```

Comment:
This field can be set to struct gdrcopy_handle.  See below.

Reply:
I don't understand the question. Do you mean embedding the struct directly here, i.e.
```
struct ofi_mr {
	struct fid_mr mr_fid;
	struct util_domain *domain;
	uint64_t key;
	uint64_t flags;
	enum fi_hmem_iface iface;
	uint64_t device;
	union {
                struct gdrcopy_handle;
        } hmem_data;
}
```

Extending this struct would be cumbersome if we want to introduce another hmem_data type right?